nengo/connection.py

```diff
@@ -374,10 +374,14 @@ def _str(self):
         if self.label is not None:
             return self.label
 
+        if self.function is not None:
+            desc = " computing '%s'" % (
+                getattr(self.function, '__name__', str(self.function)))
+        else:
+            desc = ""
```

Comment:
Generally I try to avoid `if` blocks that set a variable one way in one part of the block and a different way in another part without doing much else, because it seems like an inline if can do the same usually with more clarity. But that's just personal preference, what you have is okay.